scheduler-server/src/test/java/gov/nasa/jpl/aerie/scheduler/server/services/SchedulingDSLCompilationServiceTests.java

```diff
@@ -241,4 +241,44 @@ export default function myGoal() {
                     .anyMatch(e -> e.message().contains("TypeError: TS2322 Incorrect return type. Expected: 'Goal', Actual: 'number'."))
     );
   }
+
+  @Test
+  void testHugeGoal()
+  {
+    // This test is intended to create a Goal that is bigger than the node subprocess's standard input buffer
+    final SchedulingDSLCompilationService.SchedulingDSLCompilationResult result;
+    result = schedulingDSLCompilationService.compileSchedulingGoalDSL(
+        PLAN_ID, """
+                export default function myGoal() {
+                  return Goal.ActivityRecurrenceGoal({
+                    activityTemplate: ActivityTemplates.PeelBanana({
+                      peelDirection: 'fromStem',
+                      fancy: { subfield1: 'value1', subfield2: [{subsubfield1: 2}]},
+                      duration: 60 * 60 * 1000 * 1000 // 1 hour in microseconds
+                    }),
+                    interval: 60 * 60 * 1000 * 1000 // 1 hour in microseconds
+                  })
+                }
+            """ + " ".repeat(9001), "goalfile");
+    final var expectedGoalDefinition = new SchedulingDSL.GoalSpecifier.GoalDefinition(
+        SchedulingDSL.GoalKinds.ActivityRecurrenceGoal,
+        new SchedulingDSL.ActivityTemplate(
+            "PeelBanana",
+            Map.ofEntries(
+                Map.entry("peelDirection", SerializedValue.of("fromStem")),
+                Map.entry("fancy", SerializedValue.of(Map.ofEntries(
```

Comment:
We create a mock mission model which is imported and used here https://github.com/NASA-AMMOS/aerie/pull/144/files/0ca7153d893c52396ff60091e246a0fcac2627d9#diff-9ddb02eebf6befa5cd70d068f1189dcf09f46ccd8dd7e9329da070278105016eR106 which does have these additional fields.

Reply:
Good question, and thanks @dyst5422 for the response. I'm planning on changing the names of the activities in the mock mission model so that it doesn't pretend to be banananation and cause confusion :+1:

Reply:
Here's the change: https://github.com/NASA-AMMOS/aerie/compare/refactor/rename-mock-activities

I can add that to this PR or open a new PR, as you guys prefer.

Reply:
Seems I missed this. But there is little chance I would be the only one confused so 👍 to renaming.

Reply:
I missed this. But as I will probably not be the only one confused, 👍 to the renaming. You can put it in this PR or another, I don't mind.

Reply:
I've included this change as a commit in this PR.